packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -71,156 +71,80 @@ const defaultProps = {
   zIndex: '6000',
 };
 
-class AbstractModal extends React.Component {
-  constructor() {
-    super();
-    this.state = {
-      modalTrigger: undefined,
-    };
-    this.handleKeydown = this.handleKeydown.bind(this);
-    this.showModalDomUpdates = this.showModalDomUpdates.bind(this);
-    this.hideModalDomUpdates = this.hideModalDomUpdates.bind(this);
-    this.modalElement = React.createRef();
-  }
-
-  componentDidMount() {
+const AbstractModal = (props) => {
+  const {
+    ariaLabel,
+    children,
+    classNameModal,
+    classNameOverlay,
+    closeOnEsc,
+    closeOnOutsideClick,
+    isFullscreen,
+    isOpen,
+    role,
+    rootSelector,
+    onRequestClose,
+    zIndex,
+    ...customProps
+  } = props;
+
+  const modalElementRef = createRef();
+
+  useEffect(() => {
```

Comment:
Consider using useLayoutEffect for this. It'll execute sooner.

Reply:
Updated: https://github.com/cerner/terra-framework/pull/1041/commits/2bb1b8369609e498a558bf37c2cb085f56fb03c2#diff-b9d74312c1fdd3afc228794142f831c3R93